### ISSUE TITLE
Wrong Carplay roundabout exits ordinals

### ISSUE THREAD

**Issue by jmontane** [organicmaps/organicmaps#3569]
I tested Organic Maps with CarPlay and I detected an issue with roundabout ordinals. "2n" and "3r" forms are generated for Catalan (see attached images). Such forms are correct ordinals, but male forms. Roundabout exits are translated as "sortida", which requires feminine ordinal forms "2a" and "3a". I suspect other gendered languages, like French and Spanish, are affected too.

I guess ordinals forms are generated in run time [here](https://github.com/organicmaps/organicmaps/blob/0e0aed446ce8431fc3cd43df02ce4a5d016969f0/iphone/Maps/Classes/CarPlay/CarPlayRouter.swift#L262). I don't know if feminine ordinals forms can be generated in run time. Maybe the simple approach is to add from 1 to N strings to strings.txt (where N is max number of roundabout exits, 11?) 

![2n-sortida](https://user-images.githubusercontent.com/339645/194772425-bf18938c-8ea7-46c0-8d19-855a1f1d44f9.png)
![3r-sortida](https://user-images.githubusercontent.com/339645/194772426-8f648321-9e65-4ef3-b2cc-216100c74f8f.png)

**Comment by biodranik** [comment]
You can fix the wrong strings here: https://github.com/organicmaps/organicmaps/blob/master/data/strings/strings.txt
Search for carplay_roundabout_exit

**Comment by jmontane** [comment]
Sorry if my issue description wasn't clear enough

carplay_roundabout_exit string is correctly translated:
en = %s exit
ca = %s sortida

Issue is about how ordinals (1st, 2nd, 3rd..) are generated in runtime (%s placeholder replacement). For Catalan language 1r, 2n, 3r... forms are generated, which are wrong, because feminine forms are required: 1a, 2a, 3a,...

So, I point to add N new strings
carplay_roundabout_exit_1
en = 1st exit
carplay_roundabout_exit_2
en = 2nd exit
carplay_roundabout_exit_3
en = 3rd exit
....
carplay_roundabout_exit_N
en = Nth exit

Then a good translation can be done for every language.

**Comment by biodranik** [comment]
There are examples in strings.txt for plurals, starting from `es:one = ...` for example. All possible forms:
- zero
- one
- two
- few
- many
- other

From here: https://developer.android.com/guide/topics/resources/string-resource#Plurals

**Comment by jmontane** [comment]
These categories don't cover the case.

If %d placeholder is used at strings.txt, then all numbers (1,2,3,4...) will be replaced by male ordinals forms (primer, segon, tercer, quart,...), which is wrong.

If %d placeholder is used at strings.txt, then I need a string for each possible number (1,2,3,4...) to be able to type feminine ordinals forms (primera, segona, tercera, quarta,...). But we haven't a category for each number.

Just as example, OsmAnd has [17 ordinal strings](https://github.com/osmandapp/OsmAnd-resources/blob/master/voice/en/en_tts.js#L53), and can be [easily translated](https://github.com/osmandapp/OsmAnd-resources/blob/master/voice/es/es_tts.js#L53) for gendered languages.

**Comment by biodranik** [comment]
> If %d placeholder is used at strings.txt, then all numbers (1,2,3,4...) will be replaced by male ordinals forms (primer, segon, tercer, quart,...), which is wrong.

Sorry, I do not understand, why you can't fix it using the current approach. Why does changing current es (ca?) strings to feminine variants won't solve the problem? Are both male and female variants really needed now in the TTS?

**Comment by biodranik** [comment]
Ok, now I see the problem. The current implementation does it in the code and does it wrong for some reason (because the code supports only masculine form maybe? @AntonM030481 can you please check?):
```
      let ordinalExitNumber = NumberFormatter.localizedString(from: NSNumber(value: routeInfo.roundExitNumber),
                                                              number: .ordinal)
      let exitNumber = String(coreFormat: L("carplay_roundabout_exit"),
                              arguments: [ordinalExitNumber])
```

If it is not possible to fix it programmatically, then you're right, all these translations should be added manually into `data/strings/sounds.txt`.

**Comment by matheusgomesms** [comment]
Yep, I was about to create another issue for that. I confirm this happens also with Portuguese. It's interesting that TTS files have this separation, but not CarPlay.

Shall I add on `data/strings/strings.txt` something like this, and then only languages that this happens add translation? Or should all languages have this?

Example (only with those languages):

[carplay_roundabout_exit_1]
tags = ios
    ca = 1a sortida
    es = 1a salida
    fr = 1re sortie
    it = 1ª uscita
    pt = 1ª saída

[carplay_roundabout_exit_2]
tags = ios
    ca = 2a sortida
    es = 2a salida
    it = 2ª uscita
    pt = 2ª saída
